ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/CompactionQueryBuilder.java

```diff
@@ -203,19 +214,23 @@ CompactionQueryBuilder setIsDeleteDelta(boolean deleteDelta) {
    * @param resultTableName the name of the table we are running the operation on
    * @throws IllegalArgumentException if compactionType is null
    */
-  CompactionQueryBuilder(CompactionType compactionType, Operation operation,
+  CompactionQueryBuilder(CompactionType compactionType, Operation operation, boolean crud,
       String resultTableName) {
     if (compactionType == null) {
       throw new IllegalArgumentException("CompactionQueryBuilder.CompactionType cannot be null");
     }
+    this.compactionType = compactionType;
     this.operation = operation;
     this.resultTableName = resultTableName;
-    major = compactionType == CompactionType.MAJOR_CRUD
-        || compactionType == CompactionType.MAJOR_INSERT_ONLY;
-    crud =
-        compactionType == CompactionType.MAJOR_CRUD || compactionType == CompactionType.MINOR_CRUD;
-    minor = !major;
+    this.crud = crud;
     insertOnly = !crud;
+    major = CompactionType.MAJOR.equals(compactionType);
```

Comment:
what's the purpose of these local vars (i.e. minor, major, rebalance), why don't you use the compactionType?